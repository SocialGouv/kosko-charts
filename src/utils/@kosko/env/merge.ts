//

/* eslint-disable @typescript-eslint/no-unsafe-return */
/* eslint-disable @typescript-eslint/no-explicit-any */

// HACK(douglasduteil): wrap internal @kosko/env/dist/merge.js
// As the paths API is internal (not exported) in @kosko/env
// This is code will wrap and expose it for our needs.
// see https://github.com/tommy351/kosko/blob/%40kosko/env%401.0.1/packages/env/src/merge.ts#L12-L18
// since https://github.com/tommy351/kosko/commit/6353c8559994b826c8056ad03a35332e88b9134a

// https://stackoverflow.com/a/48769843
type UnionToIntersection<T> = (T extends any ? (k: T) => void : never) extends (
  k: infer I
) => void
  ? I
  : never;

export const merge: <T extends any[]>(
  ...data: T
) => UnionToIntersection<T[number]> =
  // eslint-disable-next-line @typescript-eslint/no-require-imports, @typescript-eslint/no-var-requires
  require("@kosko/env/dist/merge.js").merge;
