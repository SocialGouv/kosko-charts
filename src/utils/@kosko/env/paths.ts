//

import { join } from "path";

// HACK(douglasduteil): wrap internal @kosko/env/dist/paths.js
// As the paths API is internal (not exported) in @kosko/env
// This is code will wrap and expose it for our needs.
// see https://github.com/tommy351/kosko/blob/%40kosko/env%400.5.2/packages/env/src/paths.ts#L9-L13
// since https://github.com/tommy351/kosko/commit/6353c8559994b826c8056ad03a35332e88b9134a

export const formatPath: (
  path: string,
  data: Record<string, unknown>
) => string =
  // eslint-disable-next-line @typescript-eslint/no-require-imports, @typescript-eslint/no-var-requires
  require(join(require.resolve("@kosko/env"), "../paths.js")).formatPath;
