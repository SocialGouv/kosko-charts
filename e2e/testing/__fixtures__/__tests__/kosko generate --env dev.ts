//

import { getEnvManifests } from "@socialgouv/kosko-charts/testing";
import { project } from "@socialgouv/kosko-charts/testing/fake/github-actions.env";

jest.setTimeout(1000 * 60);
test("kosko generate --dev", async () => {
  expect(
    await getEnvManifests("dev", "", project("sample-next-app").dev)
  ).toMatchSnapshot();
});
