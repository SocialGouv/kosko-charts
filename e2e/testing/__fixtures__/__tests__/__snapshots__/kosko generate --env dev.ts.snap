// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kosko generate --dev 1`] = `
"---
apiVersion: v1
kind: Namespace
metadata:
  annotations:
    field.cattle.io/creatorId: gitlab
    field.cattle.io/projectId: c-bd7z2:p-7ms8p
    git/branch: master
    git/remote: >-
      https://gitlab-ci-token:[MASKED]@gitlab.factory.social.gouv.fr/SocialGouv/sample-nxt-app.git
    app.gitlab.com/app: socialgouv-sample-nxt-app
    app.gitlab.com/env: master-dev2
    app.gitlab.com/env.name: master-dev2
  labels:
    azure-pg-admin-user: sample-nxt-app
    application: master-dev2-sample-nxt-app
    owner: sample-nxt-app
    team: sample-nxt-app
    cert: wildcard
  name: sample-nxt-app-85-master-dev2
---
apiVersion: v1
kind: FooManifest
---
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.gitlab.com/app: socialgouv-sample-nxt-app
    app.gitlab.com/env: master-dev2
    app.gitlab.com/env.name: master-dev2
  labels:
    app: www
    application: master-dev2-sample-nxt-app
    owner: sample-nxt-app
    team: sample-nxt-app
    cert: wildcard
    component: nginx
  name: www
  namespace: sample-nxt-app-85-master-dev2
spec:
  replicas: 1
  selector:
    matchLabels:
      app: www
  template:
    metadata:
      annotations:
        kapp.k14s.io/disable-default-ownership-label-rules: ''
        kapp.k14s.io/disable-default-label-scoping-rules: ''
        app.gitlab.com/app: socialgouv-sample-nxt-app
        app.gitlab.com/env: master-dev2
        app.gitlab.com/env.name: master-dev2
      labels:
        app: www
        application: master-dev2-sample-nxt-app
        owner: sample-nxt-app
        team: sample-nxt-app
        cert: wildcard
        component: nginx
    spec:
      containers:
        - image: nginx:latest
          livenessProbe:
            failureThreshold: 6
            httpGet:
              path: /index.html
              port: http
            initialDelaySeconds: 30
            periodSeconds: 5
            timeoutSeconds: 5
          name: www
          ports:
            - containerPort: 80
              name: http
          readinessProbe:
            failureThreshold: 15
            httpGet:
              path: /index.html
              port: http
            initialDelaySeconds: 0
            periodSeconds: 5
            successThreshold: 1
            timeoutSeconds: 1
          resources:
            limits:
              cpu: 500m
              memory: 128Mi
            requests:
              cpu: 5m
              memory: 32Mi
          startupProbe:
            failureThreshold: 12
            httpGet:
              path: /index.html
              port: http
            periodSeconds: 5
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: www
    application: master-dev2-sample-nxt-app
    owner: sample-nxt-app
    team: sample-nxt-app
    cert: wildcard
  name: www
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.gitlab.com/app: socialgouv-sample-nxt-app
    app.gitlab.com/env: master-dev2
    app.gitlab.com/env.name: master-dev2
  namespace: sample-nxt-app-85-master-dev2
spec:
  ports:
    - name: http
      port: 80
      targetPort: 80
  selector:
    app: www
  type: ClusterIP
---
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: nginx
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.gitlab.com/app: socialgouv-sample-nxt-app
    app.gitlab.com/env: master-dev2
    app.gitlab.com/env.name: master-dev2
  labels:
    app: www
    application: master-dev2-sample-nxt-app
    owner: sample-nxt-app
    team: sample-nxt-app
    cert: wildcard
  name: www
  namespace: sample-nxt-app-85-master-dev2
spec:
  rules:
    - host: master-dev2-sample-nxt-app.dev2.fabrique.social.gouv.fr
      http:
        paths:
          - backend:
              serviceName: www
              servicePort: 80
            path: /
  tls:
    - hosts:
        - master-dev2-sample-nxt-app.dev2.fabrique.social.gouv.fr
      secretName: wildcard-crt
"
`;
